### ISSUE TITLE
Invisible text in dark mode

### ISSUE THREAD

**Issue by bootstraponline** [google/charts#258]
![image](https://user-images.githubusercontent.com/1173057/58386553-6ae71e00-7fcf-11e9-935f-e0058130c4cf.png)

See https://github.com/google/charts/issues/125

The charts are not accessible due to lack of color contrast.

**Comment by drcdev** [comment]
I've come across this as well and have been searching for a way to override or change it via theme, but have yet to find anything that works. If I find anything I'll post back here!

**Comment by drcdev** [comment]
Found some details on overriding here: https://github.com/google/charts/issues/117

Seems like it would need to be conditional to apply for each theme, since it uses the charts palette.

**Comment by bootstraponline** [comment]
Did that info work for you? The second axis is still invisible.


<img src="https://user-images.githubusercontent.com/1173057/58609381-5cba2b80-8275-11e9-9a4a-414bd62578ad.png" width="30%"/>



<summary>
<details>

```dart
/// Line chart example
/// https://google.github.io/charts/flutter/example/line_charts/area_and_line
import 'package:charts_flutter/flutter.dart' as charts;
import 'package:flutter/material.dart';

class AreaAndLineChart extends StatelessWidget {
  final List<charts.Series> seriesList;
  final bool animate;

  AreaAndLineChart(this.seriesList, {this.animate});

  /// Creates a [LineChart] with sample data and no transition.
  factory AreaAndLineChart.withSampleData() {
    return new AreaAndLineChart(
      _createSampleData(),
      // Disable animations for image tests.
      animate: false,
    );
  }

  @override
  Widget build(BuildContext context) {
    return new charts.LineChart(seriesList,
        primaryMeasureAxis: charts.NumericAxisSpec(
            renderSpec: charts.GridlineRendererSpec(
                labelStyle: charts.TextStyleSpec(
                    fontSize: 10, color: charts.MaterialPalette.white),
                lineStyle: charts.LineStyleSpec(
                    thickness: 0,
                    color: charts.MaterialPalette.gray.shadeDefault))),
        animate: animate,
        customSeriesRenderers: [
          new charts.LineRendererConfig(
              // ID used to link series to this renderer.
              customRendererId: 'customArea',
              includeArea: true,
              stacked: true),
        ]);
  }

  /// Create one series with sample hard coded data.
  static List<charts.Series<LinearSales, int>> _createSampleData() {
    final myFakeDesktopData = [
      new LinearSales(0, 5),
      new LinearSales(1, 25),
      new LinearSales(2, 100),
      new LinearSales(3, 75),
    ];

    var myFakeTabletData = [
      new LinearSales(0, 10),
      new LinearSales(1, 50),
      new LinearSales(2, 200),
      new LinearSales(3, 150),
    ];

    return [
      new charts.Series<LinearSales, int>(
        id: 'Desktop',
        colorFn: (_, __) => charts.MaterialPalette.blue.shadeDefault,
        domainFn: (LinearSales sales, _) => sales.year,
        measureFn: (LinearSales sales, _) => sales.sales,
        data: myFakeDesktopData,
      )
        // Configure our custom bar target renderer for this series.
        ..setAttribute(charts.rendererIdKey, 'customArea'),
      new charts.Series<LinearSales, int>(
        id: 'Tablet',
        colorFn: (_, __) => charts.MaterialPalette.green.shadeDefault,
        domainFn: (LinearSales sales, _) => sales.year,
        measureFn: (LinearSales sales, _) => sales.sales,
        data: myFakeTabletData,
      ),
    ];
  }
}

/// Sample linear data type.
class LinearSales {
  final int year;
  final int sales;

  LinearSales(this.year, this.sales);
}

```

</details>
</summary>

**Comment by drcdev** [comment]
You need to apply to domainAxis as well. That only has primary MeasureAxis.

I haven't tried it out yet in much detail... Will look into it tomorrow.

**Comment by bootstraponline** [comment]
That worked, I updated my comment with a complete example. Thanks!

**Comment by drcdev** [comment]
Here was my solution for a project with multiple charts and both a light and dark theme. I tried using different colors for each theme, but found that they would apply correctly on a reload, but get mixed up when toggling the theme and become inverted. I don't have the time to figure out why that won't work so I settled on grey500 as a good middle ground for both default material light/dark themes where the backgrounds haven't been changed.

I built it this way to facilitate the TimeSeriesCharts which uses different definitions for each axis. I thought about trying to cast from \<dynamic\> to \<num\> and \<DateTime\> (charts.GridlineRendererSpec returns RendererSpec\<dynamic\>), but two methods saves some time for little effort and the casting was giving me errors.

### Axis Themes
```dart
import 'package:charts_flutter/flutter.dart' as charts;

class AxisTheme {
  static charts.RenderSpec<num> axisThemeNum() {
    return charts.GridlineRendererSpec(
      labelStyle: charts.TextStyleSpec(
        color: charts.MaterialPalette.gray.shade500,
      ),
      lineStyle: charts.LineStyleSpec(
        color: charts.MaterialPalette.gray.shade500,
      ),
    );
  }

  static charts.RenderSpec<DateTime> axisThemeDateTime() {
    return charts.GridlineRendererSpec(
      labelStyle: charts.TextStyleSpec(
        color: charts.MaterialPalette.gray.shade500,
      ),
      lineStyle: charts.LineStyleSpec(
        color: charts.MaterialPalette.transparent,
      ),
    );
  }
}
```

### Applying to charts
```dart
Widget _buildChart(BuildContext context) {
    charts.RenderSpec<num> renderSpecPrimary = AxisTheme.axisThemeNum();
    charts.RenderSpec<DateTime> renderSpecDomain =
        AxisTheme.axisThemeDateTime();

    return charts.TimeSeriesChart(
      data,
      animate: true,
      defaultRenderer: new charts.LineRendererConfig(),
      dateTimeFactory: const charts.LocalDateTimeFactory(),
      primaryMeasureAxis: charts.NumericAxisSpec(
        tickProviderSpec: charts.BasicNumericTickProviderSpec(
          zeroBound: false,
        ),
        renderSpec: renderSpecPrimary,
      ),
      domainAxis: charts.DateTimeAxisSpec(
        renderSpec: renderSpecDomain,
      ),
    );
  }
```

**Comment by bootstraponline** [comment]
Thanks! I'll leave this issue open in the hopes the `charts` team makes this work out of the box in the future.

**Comment by talamaska** [comment]
All good, but this doesn't seem to work with a bar chart and ordinal spec

**Comment by fahmisatrio** [comment]
extending what @drcdev  explained earlier, i tried to add new **AxisThemeString**
### Axis Theme
```dart
import 'package:charts_flutter/flutter.dart' as charts;

class AxisTheme {
  static charts.RenderSpec<num> axisThemeNum() {
    return charts.GridlineRendererSpec(
      labelStyle: charts.TextStyleSpec(
        color: charts.MaterialPalette.white,
      ),
      lineStyle: charts.LineStyleSpec(
        color: charts.MaterialPalette.white,
      ),
    );
  }

  static charts.RenderSpec<DateTime> axisThemeDateTime() {
    return charts.GridlineRendererSpec(
      labelStyle: charts.TextStyleSpec(
        color: charts.MaterialPalette.white,
      ),
      lineStyle: charts.LineStyleSpec(
        color: charts.MaterialPalette.transparent,
      ),
    );
  }

  static charts.RenderSpec<String> axisThemeString() {
    return charts.SmallTickRendererSpec(
      labelStyle: new charts.TextStyleSpec(
        color: charts.MaterialPalette.white,
      ),
      lineStyle: charts.LineStyleSpec(
        color: charts.MaterialPalette.transparent,
      ),
    );
  }
}

```

### Applying to Bar Chart and Ordinal Spec

```dart
Widget _buildChart(BuildContext context) {
    charts.RenderSpec<String> renderSpecDomain =
        AxisTheme.axisThemeString();

    return charts.BarChart(
      data,
      animate: true,
      domainAxis: charts.OrdinalAxisSpec(
        renderSpec: renderSpecDomain,
      ),
    );
  }
```

thanks, this thread saved my day
